src/App.js

```diff
@@ -14,11 +14,108 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+let selectedGoodsArr = [];
+
+export class App extends React.Component {
+  state = {
+    selectedGoodName: '',
+  }
+
+  separator = ' ----- ';
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          Selected good:
+          {this.state.selectedGoodName}
+        </h1>
+
+        <button
+          type="button"
+          onClick={(event) => {
+            selectedGoodsArr = [];
+            this.setState(state => ({
+              selectedGoodName: selectedGoodsArr
+                .map(good => good
+                  .innerText
+                  .split(this.separator)[0])
+                .join(', '),
+            }));
+            const items = [...document.body.querySelectorAll('.selected')];
```

Comment:
don't use DOM

Reply:
use condition inside className for classes and use the state to save all data that you need